hax/hax/ha/ha.py

```diff
@@ -55,10 +58,36 @@ def check_and_send(self,
                 self.util.is_proc_local(fid)):
             resource = self.util.get_process_node(fid)
             resource_id = str(Conf.machine_id)
-            logging.debug('Sending %s event for resource %s',
-                          resource_status, resource)
+            LOG.debug('Sending %s event for resource %s',
+                      resource_status, resource)
             self.send_event(
                 resource_type=parent_resource_type,
                 resource_id=resource_id,
                 resource_name=resource,
                 resource_status=resource_status)
+
+    def broadcast(self, ha_states: List[HAState]) -> None:
+        cns = self.util
+        LOG.debug('Notifying HA with states %s', ha_states)
+        for st in ha_states:
+            if st.status != ObjHealth.OFFLINE:
+                continue
+
+            if st.fid.container == ObjT.PROCESS.value:
```

Comment:
What if ha.broadcast is not invoked for ObjT.PROCESS type. We need to still send and event to HA right?

Reply:
Right now we are sending only node events to HA, which we are generating based on processes state after aggregation.
So does the Jira describes.

For other ObjT types such as drives, enclosure -> what kind of events are we supposed to send? Could you please explain.

Reply:
Refactored the code as dicussed.